### PR TITLE
[FEAT] : priority and Filters functionality added

### PR DESCRIPTION
**Pull request by t007rushi**
- Priority added 
- 
FILTER BY : 
- Label 
- Priority 

SORT BY : 
- Time/Date

recording

<details><summary>Browser metadata</summary>
```
Path:      /search
Browser:   Chrome 99.0.4844.84 on Windows 10
Viewport:  1536 x 775 @1.25x
Language:  en-US
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-12--monkeep.netlify.app%2Fsearch&os=Windows&os_version=10&browser=Chrome&browser_version=99.0&resolution=1536x775&speed=1&start=true&ref=netlify-source)

</details>

[Open Deploy Preview](https://deploy-preview-12--monkeep.netlify.app/search) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=624ea144f677890008f349c7&commentID=624ea33a4bba3e4c95a84412&resolution=resolved)

### CODE REVIEW

File: src/Layout/Header/Header.jsx [hunk 1]
```diff
@@ -25,14 +27,21 @@ export const Header = () => {
...
         </NavLink>
       </div>
-      {user.isUserLoggedIn && (
+      {user.isUserLoggedIn &&  (
+        <div className={ showSearch?"search-active":"search-hide"}>
         <div className="flex-row header-mid">
-          <AiOutlineSearch className="header-icon" />
+        <NavLink to="/notes"> <BiArrowBack className="header-icon" onClick={() =>{
+            hideSearchbar();
+          }} /></NavLink>
```
Comment (JantuDeb) [inline src/Layout/Header/Header.jsx]: `onCLick={hideSearchbar}`

File: src/Layout/Header/Header.jsx [hunk 2]
```diff
@@ -25,14 +27,21 @@ export const Header = () => {
...
           <NavLink to="/search"> <input
             type="text"
             placeholder="Search"
             className="search-bar"
           /></NavLink>
+          <NavLink to="/search"> <IoMdOptions className="filter-icon" onClick={() => {
+            ShowFilters();
+          }}/></NavLink>
```
Comment (JantuDeb) [inline src/Layout/Header/Header.jsx]: same here

File: src/components/CreateNote/CreateNote.jsx [hunk 3]
```diff
@@ -30,86 +33,108 @@ export const CreateNote = () => {
...
+            e.preventDefault();
+            closerEditor();
+            addNotes({
+              ...notesData,
+              description: description,
+              inTrash: false,
+            });
+            noteDispatcher({ type: CLEAR_INPUTS, payload: initialState });
+            setDescription("");
+          }}
```
Comment (JantuDeb) [inline src/components/CreateNote/CreateNote.jsx:49]: You can create a function for that

**Review by bossmonkey98 — APPROVED** [review]
Everything's fine Good to merge

**Review by JantuDeb — APPROVED** [review]
This is looks cool, added a suggestion you can go through this

**Review by Hamdrive — COMMENTED** [review]
Nice implementation of context and reducer to handle notes filters, also use of currying concept is bonus.
Also good use of abstraction and barrel imports.
Suggestion would be to consider having less context's, can be looked into for future PRs

**Review by abhijeetekad — APPROVED** [review]
Code looks good, learned a lot,
- I like how you have imported moment and used that to show timing, also it is dynamic 🔥 .
- Compose function for note filter working fine and code is also clean and neat.
- great work

File: src/components/CreateNote/CreateNote.jsx (hunk 3 above)
Comment (t007rushi) [reply src/components/CreateNote/CreateNote.jsx:49]: Thank YOu pointing outwill do that

**Pull request merged**